src/sections/trade/sections/swap/SwapPage.tsx

```diff
@@ -65,11 +65,14 @@ export function SwapPage() {
   const [addToken, setAddToken] = useState(false)
   const [tokenCheck, setTokenCheck] = useState<Asset | null>(null)
 
-  const isEvm = isEvmAccount(account?.address)
-  const version = useRemount([isEvm])
-
   const rpcUrlList = useActiveRpcUrlList()
   const activeProvider = useActiveProvider()
+  const isTestnet =
```

Comment:
I'm not sure this will work in auto mode. You are comparing only to first RPC in the list, but it could connect to 2nd or 3rd.

I believe it should be fixed in `useActiveProvider()` hook at `src/api/provider.ts:324` so it returns correct provider with correct dataEnv.

Reply:
it should work, since, we just need if an rpc is mainnet or testnet. And with auto mode all rps are mainnet. This fixes an issue for testnet when the useActiveProvider return a default env values, meaning that if you are on testnet but connected to the mainnet rpc, default value will be applied as first. And SwapPage won't change its configuration even if isTestnet value is changed